modules/bidglassBidAdapter.js

```diff
@@ -0,0 +1,187 @@
+    server needs.
+    */
+
+    /*
+    Sample array entry for validBidRequests[]:
+    [{
+      "bidder": "bidglass",
+      "bidId": "51ef8751f9aead",
+      "params": {
+      "adUnitId": 11,
+      ...
+      },
+      "adUnitCode": "div-gpt-ad-1460505748561-0",
+      "transactionId": "d7b773de-ceaa-484d-89ca-d9f51b8d61ec",
+      "sizes": [[320,50],[300,250],[300,600]],
+      "bidderRequestId": "418b37f85e772c",
+      "auctionId": "18fd8b8b0bd757",
+      "bidRequestsCount": 1
+    }]
+    */
+
+    let imps = [];
+    let getReferer = function() {
+      if (window === window.top) {
+        return window.location.href;
+      } else if (window.parent === window.top) {
+        return document.referrer;
+      } else {
+        return null;
+      }
+    };
+    let getOrigins = function() {
+      var ori = [window.location.protocol + '//' + window.location.hostname];
+
+      if (window.location.ancestorOrigins) {
+        for (var i = 0; i < window.location.ancestorOrigins.length; i++) {
+          ori.push(window.location.ancestorOrigins[i]);
+        }
+      } else if (window !== window.top) {
+        // Derive the parent origin
+        var parts = document.referrer.split('/');
+
+        ori.push(parts[0] + '//' + parts[2]);
+
+        if (window.parent !== window.top) {
+          // Additional unknown origins exist
+          ori.push('null');
+        }
+      }
+
+      return ori;
+    };
+
+    utils._each(validBidRequests, function(bid) {
+      bid.sizes = ((utils.isArray(bid.sizes) && utils.isArray(bid.sizes[0])) ? bid.sizes : [bid.sizes]);
+      bid.sizes = bid.sizes.filter(size => utils.isArray(size));
+
+      // Stuff to send: [bid id, sizes, adUnitId]
+      imps.push({
+        bidId: bid.bidId,
+        sizes: bid.sizes,
+        adUnitId: utils.getBidIdParameter('adUnitId', bid.params)
+      });
+    });
+
+    // Stuff to send: page URL
+    const bidReq = {
+      reqId: utils.getUniqueIdentifierStr(),
+      imps: imps,
+      ref: getReferer(),
+      ori: getOrigins()
+    };
+
+    let url = 'https://bid.glass/ad/hb.php?' +
+      `src=$$REPO_AND_VERSION$$`;
+
+    return {
+      method: 'POST',
+      url: url,
+      data: JSON.stringify(bidReq),
+      options: {
+        contentType: 'text/plain',
+        withCredentials: false
+      }
+    }
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {ServerResponse} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse) {
+    // const serverBody  = serverResponse.body;
+    // const headerValue = serverResponse.headers.get('some-response-header');
+
+    const bidResponses = [];
+
+    utils._each(serverResponse.body.bidResponses, function(bid) {
+      bidResponses.push({
+        requestId: bid.requestId,
+        cpm: parseFloat(bid.cpm),
+        width: parseInt(bid.width),

```

Comment:
parseInt takes two parameters string and radix. I would suggest to add radix as it does not default to 10 and may return unexpected results. 
More info here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt and interesting article on parseInt https://medium.com/dailyjs/parseint-mystery-7c4368ef7b21

Reply:
Thanks, good call.